### PR TITLE
feat. add other-plugins-events attribute

### PR DESCRIPTION
**Pull request by nivcoo**
Hello, i've added attribut to call event of specific plugin depending on the item. Let me know if you want to change anything

### CODE REVIEW

**Comment by OmerBenGera** [comment]
I told you whats the issue with that in the ticket.
Closing this for now, you are more than welcome to make a new PR with another implementation.
Thanks for the time spending on this tho :)

**Comment by nivcoo** [comment]
Have you see my comment about this commit ?

**Comment by OmerBenGera** [comment]
I will look into the code when I am at home.
Thanks for contributing!

**Comment by nivcoo** [comment]
> I will look into the code when I am at home.
> Thanks for contributing!

Thx, there is surely some optimization to do but the idea behind is present :p

**Comment by nivcoo** [comment]
I've move the load of otherEvents to be only load once

File: src/main/java/com/bgsoftware/wildtools/handlers/EventsHandler.java [hunk 1]
```diff
@@ -68,7 +79,20 @@ public void loadClaimingPlugins(List<String> claimingPlugins){
...
+        for (Tool tool : tools) {
+            otherPluginsBreakMethodsTools.put(tool, new ArrayList<>());
+            List<CachedListenerMethod> otherPluginsMethods = otherPluginsBreakMethodsTools.get(tool);
+            List<String> plugins = new ArrayList<>(tool.otherPluginsEvents());
+            plugins.addAll(otherPlugins);
+            loadOtherPlugins(plugins, otherPluginsMethods);
+        }
+    }
+
+    public void loadOtherPlugins(List<String> otherPlugins, List<CachedListenerMethod> otherPluginsBreakMethods) {
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/wildtools/handlers/EventsHandler.java:94]: Is otherPluginsBreakMethods necessary for the method?

File: src/main/java/com/bgsoftware/wildtools/objects/tools/WTool.java [hunk 2]
```diff
@@ -350,6 +356,9 @@ public boolean hasAnvilCombineLimit() {
         return new HashSet<>(whitelistedDrops);
     }
 
+    @Override
+    public Set<String> otherPluginsEvents() { return otherPluginsEvents; }
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/wildtools/objects/tools/WTool.java:360]: Don't use inline methods.

**Review by OmerBenGera — CHANGES_REQUESTED** [review]

File: src/main/java/com/bgsoftware/wildtools/handlers/EventsHandler.java [hunk 3]
```diff
@@ -68,7 +79,20 @@ public void loadClaimingPlugins(List<String> claimingPlugins){
         claimingPluginsInteractMethods.sort(CachedListenerMethod::compareTo);
     }
 
-    public void loadOtherPlugins(List<String> otherPlugins){
+    public void loadOtherPlugins(List<String> otherPlugins) {
+        otherPluginsBreakMethodsTools.clear();
+        List<Tool> tools = plugin.getToolsManager().getTools();
+        for (Tool tool : tools) {
+            otherPluginsBreakMethodsTools.put(tool, new ArrayList<>());
+            List<CachedListenerMethod> otherPluginsMethods = otherPluginsBreakMethodsTools.get(tool);
```
[the remark this answers is not in the document]
Comment (nivcoo) [reply src/main/java/com/bgsoftware/wildtools/handlers/EventsHandler.java:87]: I don't understand exactly what you mean here

**Comment by nivcoo** [comment]
Any news ?

**Comment by nivcoo** [comment]
Any news ?

**Comment by nivcoo** [comment]
Any news ?

**Comment by OmerBenGera** [comment]
> Any news ?

I totally forgot about that, I am so sorry you had to wait that long.
I will look into that tomorrow.

**Comment by nivcoo** [comment]
Thx

**Review by OmerBenGera — APPROVED** [review]

**Review by OmerBenGera — CHANGES_REQUESTED** [review]
Should be merged to dev, not to master

**Comment by nivcoo** [comment]
Can you check my last changes for conflicts ? i think it's good

**Comment by nivcoo** [comment]
Sorry for lot of commit :/ let me know if it's good idk how works merge like that :/

**Comment by nivcoo** [comment]
Any news ?

**Pull request merged**